pkg/apiserver/resthandler.go

```diff
@@ -684,15 +686,29 @@ func DeleteResource(r rest.GracefulDeleter, checkBody bool, scope RequestScope,
 // to use it.
 // TODO: add appropriate structured error responses
 func queryToObject(query url.Values, scope RequestScope, kind string) (runtime.Object, error) {
```

Comment:
I think ServerAPIVersion is wrong. Groups should be using query options registered to their own group/version, or unversioned query options, but not ones from v1